tests/rest/client/utils.py

```diff
@@ -116,6 +119,10 @@ def create_room_as(
                 Note that if is_public is set, the "visibility" key will be overridden.
                 If room_version is set, the "room_version" key will be overridden.
             custom_headers: HTTP headers to include in the request.
+            appservice_user_id: The `user_id` URL parameter to pass.
+                This allows driving an application service user
+                using an application service access token in `tok`.
+
```

Comment:
as a sidebar: I kinda hate these `do_foo_as` methods which mess about with the auth mechanism: I find them harder to use and understand than just doing the request. But I'll not insist on a rewrite here.

Reply:
I backed out these changes. 👍